### PR TITLE
WIP mongodb api implementation

### PR DESCRIPTION
**Pull request by willemodendaal**
Still a work in progress, so not merging this yet.

Feel free to comment, I'll update this PR as I progress.

### CODE REVIEW

File: src/API/Loggy.Api/Loggy.Api.csproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.2</TargetFramework>
+    <UserSecretsId>dcad0434-e2d3-49d0-ae7d-e3630e0ddbc7</UserSecretsId>
```
Comment (JPThorne) [inline src/API/Loggy.Api/Loggy.Api.csproj:5]: +1 for user secrets

File: src/API/Loggy.Api/Loggy.Api.csproj [hunk 2]
```diff
@@ -14,6 +14,7 @@
     <PackageReference Include="Microsoft.AspNetCore.App" />
     <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.2.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="2.2.0" />
+    <PackageReference Include="MongoDB.Driver" Version="2.7.3" />
   </ItemGroup>
 
 </Project>
```
Comment (JPThorne) [inline src/API/Loggy.Api/Loggy.Api.csproj:26]: Just a thought on this - would it not maybe make sense to have a Loggy.Data (DataAccess / Db ?) Project, and have that project have a nice clean interface, and it in turn references some sort of Loggy.Data.Mongo project?  And of course then that Loggy.Data.Mongo project references Mongo specific nuget packages. 

I think that might be cleaner and it gives us flexibility with the scenario where we decide "actually mongo has some issues and so we want to switch to ... sql (for example)" 

What are your thoughts @willemodendaal ?

Comment (willemodendaal) [reply src/API/Loggy.Api/Loggy.Api.csproj:26]: Sure, that's worth considering. I've felt pains with too-tight coupling to the db layer before. Looking into this now.

Comment (JPThorne) [reply src/API/Loggy.Api/Loggy.Api.csproj:26]: coolbeans :) looking forward to seeing how it goes

**Review by RJMcCartney — COMMENTED** [review]
models and GraphQl schema ! Yey 🥇 💃

**Pull request merged**